// @fluentui/react-button/lib-commonjs/components/MenuButton/useMenuButtonStyles.styles.raw.headless.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    menuButtonClassNames: function() {
        return menuButtonClassNames;
    },
    useMenuButtonStyles_unstable: function() {
        return useMenuButtonStyles_unstable;
    }
});
const menuButtonClassNames = {
    root: 'fui-MenuButton',
    icon: 'fui-MenuButton__icon',
    menuIcon: 'fui-MenuButton__menuIcon'
};
const useMenuButtonStyles_unstable = (state)=>{
    'use no memo';
    const { appearance, disabled, disabledFocusable, shape, size, icon, iconOnly } = state;
    const expanded = !!state.root['aria-expanded'];
    state.root.className = [
        menuButtonClassNames.root,
        // Appearance
        appearance && `${menuButtonClassNames.root}--${appearance}`,
        // Size
        size && `${menuButtonClassNames.root}--${size}`,
        // Shape
        shape && `${menuButtonClassNames.root}--${shape}`,
        // Disabled styles
        disabled && `${menuButtonClassNames.root}--disabled`,
        disabledFocusable && `${menuButtonClassNames.root}--disabledFocusable`,
        // Expanded
        expanded && `${menuButtonClassNames.root}--expanded`,
        // Icons
        icon && iconOnly && `${menuButtonClassNames.root}--iconOnly`,
        // User provided class name
        state.root.className
    ].filter(Boolean).join(' ');
    if (state.icon) {
        state.icon.className = [
            menuButtonClassNames.icon,
            state.icon.className
        ].filter(Boolean).join(' ');
    }
    if (state.menuIcon) {
        state.menuIcon.className = [
            menuButtonClassNames.menuIcon,
            state.menuIcon.className
        ].filter(Boolean).join(' ');
    }
    return state;
};
