// @fluentui/react-button/lib-commonjs/components/SplitButton/useSplitButtonStyles.styles.raw.headless.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    splitButtonClassNames: function() {
        return splitButtonClassNames;
    },
    useSplitButtonStyles_unstable: function() {
        return useSplitButtonStyles_unstable;
    }
});
const splitButtonClassNames = {
    root: 'fui-SplitButton',
    menuButton: 'fui-SplitButton__menuButton',
    primaryActionButton: 'fui-SplitButton__primaryActionButton'
};
const useSplitButtonStyles_unstable = (state)=>{
    'use no memo';
    const { appearance, disabled, disabledFocusable, shape, size } = state;
    state.root.className = [
        splitButtonClassNames.root,
        // Appearance
        appearance && `${splitButtonClassNames.root}--${appearance}`,
        // Size
        size && `${splitButtonClassNames.root}--${size}`,
        // Shape
        shape && `${splitButtonClassNames.root}--${shape}`,
        // Disabled styles
        disabled && `${splitButtonClassNames.root}--disabled`,
        disabledFocusable && !disabled && `${splitButtonClassNames.root}--disabledFocusable`,
        // User provided class name
        state.root.className
    ].filter(Boolean).join(' ');
    if (state.primaryActionButton) {
        state.primaryActionButton.className = [
            splitButtonClassNames.primaryActionButton,
            state.primaryActionButton.className
        ].filter(Boolean).join(' ');
    }
    if (state.menuButton) {
        state.menuButton.className = [
            splitButtonClassNames.menuButton,
            state.menuButton.className
        ].filter(Boolean).join(' ');
    }
    return state;
};
